src/js/components/MainNav/MainNav.js

```diff
@@ -1,30 +1,95 @@
 import React, { useContext } from 'react';
-import { withRouter } from 'react-router';
 import '../../../scss/components/MainNav/MainNav.scss';
 import { GlobalContext } from '../../contexts/GlobalContextWrapper';
 import AddServerButton from '../AddServerButton/AddServerButton';
 import MainNavButton from '../MainNavButton/MainNavButton';
 import ServerLink from '../ServerLink/ServerLink';
+import useDiscordData from '../../hooks/useDiscordData';
 
-/**
- * Let's wrap the component with withRouter() to make it receptive to route changes
- */
-const MainNav = withRouter(() => (
-  <div className="nav-column nav-column-server">
-    <div className="nav-group">
-      <MainNavButtons />
+const MainNav = () => {
+  const { Config } = useContext(GlobalContext);
+
+  const config = Config.get([
+    'discordAPIResources',
+    'mainNavButtons',
+    'mainNavButtonTypes',
+    'discordUrls',
+  ]);
+
+  const { discordAPIResources, mainNavButtons, mainNavButtonTypes, discordUrls } = config;
+  const { link, addServerButton } = mainNavButtonTypes;
+  const { client, guilds, icons } = discordAPIResources;
+  const { appCDN } = discordUrls;
+
+  const guildData = useDiscordData(guilds, client);
+
+  let addGuildData;
+
+  if (guildData) {
```

Comment:
What happens if there isn't guild data?

Reply:
The main nav will render with just the home and utility buttons